csrc/serde/fusion_cache.fbs

```diff
@@ -188,8 +206,91 @@ table ScalarInput {
   dtype: DataType;
 }
 
+// =====================================================================================
+// Tables for ArgAbstract, PhiloxCudaState, ScalarCpu, TensorArg, KernelArgumentHolder used in FusionExecutor.
+
+// Data of PhiloxCudaState
+table PhiloxCudaState {
+  seed : ulong;
+  offset : ulong;
+}
+
+// The ScalarCpu is represented by a fixed size array of raw bytes.
+table ScalarCpu {
+  instance : [byte];
+  size : ulong;
+}
+
+// Data of TensorArg.
+// The original cpp TensorArg holds real data.
+// However, only a metadata tensor is returned upon deserialization.
+// The ptr parameter is used to determine vectorization during scheduling.
+table TensorArg {
+  ptr : ulong;
+  sizes : [long];
+  strides : [long];
+  dtype: DataType;
+  is_int_index_mode : bool;
+  index_type_resolved : bool;
+}
+
+// This table corresponds with a given ArgAbstract object.
+table ArgAbstract {
+  data : ArgAbstractData;
+}
+
+// This table holds multiple ArgAbstract objects.
+table KernelArgumentHolder {
+  arguments : [ArgAbstract];
+  device_index : byte;
+  cache_id : ulong;
+}
+
 //
 // =====================================================================================
+// Tables for LaunchParams, GlobalBufferInfo, ExecutorEntry used in FusionExecutor
+
+// This table holds the cached launch parameters for a kernel.
+table LaunchParams {
+  gdimx : long;
```

Comment:
do we need long here? I thought int would be enough... https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#features-and-technical-specifications

Reply:
oh, NVM, I realized that this is merely a mapping to what we have in LaunchParams defined in executor_params.h.